### PR TITLE
[Doc]--Add negative acknowledgement to python client docs.

### PR DESCRIPTION
**Pull request by Huanli-Meng**
Add negative acknowledgement contents to python clients for the following releases:
master
2.5.2,
2.5.1,
2.5.0
2.4.2
2.4.1
2.4.0

*(If this PR fixes a github issue, please add `Fixes #<xyz>`.)*

Fixes #<xyz>

*(or if this PR is one task of a github issue, please add `Master Issue: #<xyz>` to link to the master issue.)*

Master Issue: #<xyz>

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve.*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

  - Does this pull request introduce a new feature? ( no)
  - If yes, how is the feature documented? ( docs)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

**Comment by Huanli-Meng** [comment]
@zymap , could you please help review it?

File: site2/website/versioned_docs/version-2.4.0/client-libraries-python.md [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+Installation via PyPi is available for the following Python versions:
+
+Platform | Supported Python versions
+:--------|:-------------------------
+| MacOS<br /> 10.11 (El Capitan) — 10.12 (Sierra) — 10.13 (High Sierra) — 10.14 (Mojave)<br />| 2.7, 3.7
+|Linux | 2.7, 3.4, 3.5, 3.6, 3.7
+
+### Installing from source
+
+To install the `pulsar-client` library by building from source, follow [these instructions](client-libraries-cpp.md#compilation) and compile the Pulsar C++ client library. That will also build the Python binding for the library.
```
Comment (Anonymitaet) [inline site2/website/versioned_docs/version-2.4.0/client-libraries-python.md:31]:
```suggestion
To install the `pulsar-client` library by building from source, follow [these instructions](client-libraries-cpp.md#compilation) and compile the Pulsar C++ client library. That also builds the Python binding for the library.
```

File: site2/website/versioned_docs/version-2.4.0/client-libraries-python.md [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+$ sudo python setup.py install
+```
+
+## API Reference
+
+The complete Python API reference is available at [api/python](/api/python).
+
+## Examples
+
+Below you'll find a variety of Python code examples for the `pulsar-client` library.
```
Comment (Anonymitaet) [inline site2/website/versioned_docs/version-2.4.0/client-libraries-python.md:47]:
```suggestion
You can find a variety of Python code examples for the `pulsar-client` library as below.
```

File: site2/website/versioned_docs/version-2.4.0/client-libraries-python.md [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+
+for i in range(10):
+    producer.send(('Hello-%d' % i).encode('utf-8'))
+
+client.close()
+```
+
+### Consumer example
+
+This creates a consumer with the `my-subscription` subscription on the `my-topic` topic, listen for incoming messages, print the content and ID of messages that arrive, and acknowledge each message to the Pulsar broker:
```
Comment (Anonymitaet) [inline site2/website/versioned_docs/version-2.4.0/client-libraries-python.md:68]:
```suggestion
This example creates a consumer with the `my-subscription` subscription on the `my-topic` topic, listens for incoming messages, prints the content and ID of messages that arrive, and acknowledges each message to the Pulsar broker:
```

File: site2/website/versioned_docs/version-2.4.0/client-libraries-python.md [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+
+```python
+producer = client.create_producer(
+                    topic='my-topic',
+                    schema=AvroSchema(Example) )
+
+producer.send(Example(a='Hello', b=1))
+```
+
+When the producer is created, the Pulsar broker will validate that
```
Comment (Anonymitaet) [inline site2/website/versioned_docs/version-2.4.0/client-libraries-python.md:165]:
```suggestion
When the producer is created, the Pulsar broker validates that
```

File: site2/website/versioned_docs/version-2.4.0/client-libraries-python.md [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+the existing topic schema is indeed of "Avro" type and that the
+format is compatible with the schema definition of the `Example`
+class.
+
+If there is a mismatch, the producer creation will raise an
```
Comment (Anonymitaet) [inline site2/website/versioned_docs/version-2.4.0/client-libraries-python.md:170]:
```suggestion
If there is a mismatch, the producer creation raises an
```

File: site2/website/versioned_docs/version-2.4.0/client-libraries-python.md [hunk 6]
```diff
@@ -0,0 +1,283 @@
...
+exception.
+
+Once a producer is created with a certain schema definition,
+it will only accept objects that are instances of the declared
```
Comment (Anonymitaet) [inline site2/website/versioned_docs/version-2.4.0/client-libraries-python.md:174]:
```suggestion
it only accepts objects that are instances of the declared
```

File: site2/website/versioned_docs/version-2.4.0/client-libraries-python.md [hunk 7]
```diff
@@ -0,0 +1,283 @@
...
+schema class.
+
+Similarly, for a consumer/reader, the consumer will return an
```
Comment (Anonymitaet) [inline site2/website/versioned_docs/version-2.4.0/client-libraries-python.md:177]:
```suggestion
Similarly, for a consumer/reader, the consumer returns an
```

File: site2/website/versioned_docs/version-2.4.0/client-libraries-python.md (hunk 6 above)
Comment (Anonymitaet) [reply site2/website/versioned_docs/version-2.4.0/client-libraries-python.md:174]: pls check all

**Comment by Huanli-Meng** [comment]
close the PR as CI test fails and cannot fix it. Re-create a new PR for update.

**Pull request closed without merging**